### PR TITLE
Document ESQL's csv-spec files

### PR DESCRIPTION
**Pull request by nik9000**

### CODE REVIEW

**Review by costin — APPROVED** [review]
❤️ LGTM

**Review by luigidellaquila — APPROVED** [review]
❤️

File: x-pack/plugin/esql/qa/testFixtures/src/main/resources/README.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+;
+
+// tag::sin-result[]
+a:double | sin:double
+     1.8 | 0.9738476308781951
+// end::sin-result[]
+;
+```
+
+The `// tag::` and `// end::` are standard asciidoc syntax. Weird looking but
```
Comment (leemthompo) [inline x-pack/plugin/esql/qa/testFixtures/src/main/resources/README.md:137]:
```suggestion
The `// tag::` and `// end::` are standard asciidoc syntax for working with [tagged regions](https://docs.asciidoctor.org/asciidoc/latest/directives/include-tagged-regions/#tagging-regions). Weird looking but
```

File: x-pack/plugin/esql/qa/testFixtures/src/main/resources/README.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    description = "Returns ths {wikipedia}/Sine_and_cosine[Sine] trigonometric function of an angle.",
+    examples = @Example(file = "floats", tag = "sin")
+)
+```
+
+Running the tests will generate the asciidoc files for you. See
+`esql/functions/README.md` for all of the docs the tests generate.
+
+Either way, the docs need four special comments in the CSV-SPEC files to find
+the bits to extract:
```
Comment (leemthompo) [inline x-pack/plugin/esql/qa/testFixtures/src/main/resources/README.md:120]:
```suggestion
Either way, CSV-SPEC files must be tagged using four special comments so snippets can be
included in the docs:
```

File: x-pack/plugin/esql/qa/testFixtures/src/main/resources/README.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+[source.merge.styled,esql]
+----
+include::{esql-specs}/floats.csv-spec[tag=sin]
+----
+[%header.monospaced.styled,format=dsv,separator=|]
+|===
+include::{esql-specs}/floats.csv-spec[tag=sin-result]
+|===
+```
+
```
Comment (leemthompo) [inline x-pack/plugin/esql/qa/testFixtures/src/main/resources/README.md:102]:
```suggestion
<details>
  <summary>What is this asciidoc syntax?</summary>

The first section is a source code block for the ES|QL query: 

- a [source](https://docs.asciidoctor.org/asciidoc/latest/verbatim/source-blocks/) code block (delimited by `----`)
	- `source.merge.styled,esql` indicates custom syntax highlighting for ES|QL
- an [include directive](https://docs.asciidoctor.org/asciidoc/latest/directives/include/) to import content from another file into the current document
- file path defined as an [attribute](https://docs.asciidoctor.org/asciidoc/latest/attributes/document-attributes/) or variable, within `{}`

The second section is the response returned as a table:

- styled using `[%header.monospaced.styled,format=dsv,separator=|]`
- delimited by `|===`
- again using includes, attributes, and tagged regions
</details>

```

Comment (leemthompo) [reply x-pack/plugin/esql/qa/testFixtures/src/main/resources/README.md:102]: This is a nice opportunity to explain the asciidoc syntax, and using a [collapsible](https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab) here so it doesn't take up too much space in readme but is available for folks who're interested :)

File: x-pack/plugin/esql/qa/testFixtures/src/main/resources/README.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+The second-simplest way to run the CSV-SPEC tests is to run `EsqlSpecIT` in
+`:x-pack:plugin:esql:qa:server:single-node` via the Gradle runner in IntelliJ
+or on the command line. That will boot a real Elasticsearch node, create some
+test data, and run the tests. The tests are reused in a few more scenarios,
+include multi-node and mixed-cluster.
+
+## Organization
+
+The CSV-SPEC tests grew organically for a long time, but we've since grown
+general organizing principals. But lots of tests don't follow those principals.
```
Comment (leemthompo) [inline x-pack/plugin/esql/qa/testFixtures/src/main/resources/README.md:27]:
```suggestion
general organizing principles. But lots of tests don't follow those principles.
```

🤓

**Review by leemthompo — COMMENTED** [review]
This is fantastic ❤️ , just a few additional suggestions to explain some of the asciidoc scariness.

**Comment by nik9000** [comment]
Thanks friends!

**Pull request merged**